src/components/CreateNewVM.vue

```diff
@@ -0,0 +1,342 @@
+        value: this.newProgram.code
+      }
+      if (selectedLanguage.code !== 'undefined') {
+        cmOption.value = selectedLanguage.code
+      }
+      return cmOption
+    },
+
+    addVolume (isPersistent) {
+      if (isPersistent) {
+        this.newProgram.volumes.push({
+          id: (Math.random() + 1).toString(36).substring(7),
+          comment: '',
+          isPersistent: true,
+          persistence: 'host'
+        })
+      } else {
+        this.newProgram.volumes.push({
+          id: (Math.random() + 1).toString(36).substring(7),
+          comment: '',
+          isPersistent: false,
+          use_latest: false
+        })
+      }
+    },
+    deleteVolume (volume) {
+      this.newProgram.volumes.splice(this.newProgram.volumes.findIndex(v => v.id === volume.id), 1)
+    },
+    backStep (step) {
+      this.step = step
+    },
+    async putContent (message, content, inline_requested, storage_engine, api_server) {
+      let inline = inline_requested
+      if (inline) {
+        let serialized = JSON.stringify(content)
+        if (serialized.length > 150000) {
+          inline = false
+        } else {
+          message.item_type = 'inline'
+          message.item_content = serialized
+          message.item_hash = new shajs.sha256().update(serialized).digest('hex')
+        }
+      }
+      if (!inline) {
+        let hash = ''
+        if (storage_engine === 'ipfs') {
+          message.item_type = 'ipfs'
+          hash = await ipfs_push(content, { api_server: api_server })
+        } else {
+          message.item_type = 'storage'
+          hash = await storage_push(content, { api_server: api_server })
+        }
+        message.item_hash = hash
+      }
+    },
+    async importCode () {
+      this.loading = true
+      const zip = new JSZip()
+      zip.file('main.py', this.newProgram.code)
+      this.exportFile = await zip.generateAsync({ type: 'blob' })
+      this.step = 2
+      this.loading = false
+    },
+    async send (message) {
+      if (this.account.type === 'NULS') {
+        message = await nuls.sign(Buffer.from(this.account.private_key, 'hex'), message)
+      } else if (this.account.type === 'NULS2') {
+        message = await nuls2.sign(this.account, message)
+      } else if (this.account.type === 'NEO') {
+        message = await neo.sign(this.account, message)
+      } else if (this.account.type === 'ETH') {
+        message = await ethereum.sign(this.account, message)
+      }
+
+      await broadcast(message, { api_server: this.api_server })
+    },
+    async deploy () {
+      this.loading = true
+      let storeMessage = await store.submit(
+        this.account.address,
+        {
+          chain: 'ETH',
+          sender: this.account.address,
+          account: this.account,
+          channel: 'TEST',
+          fileobject: this.exportFile,
+          api_server: this.api_server
+        }
+      ).catch((response) => {
+        this.loading = false
+      })
```

Comment:
What happends when the backend is not available ?

Reply:
The 'Deploy' button will stop loading and reactivate again.